### DISCUSSION TITLE
deta drive , how to list folder ?

### DISCUSSION THREAD

**Discussion by esaid** [deta/cloud-docs#698] [Help]
Dear all,
do you have a solution to list folders deta.Drive ?

cheers

Emmanuel

**Comment by begoon** [comment]
If you mean to list "the folders" in a particular drive with a given name, for example, "MyFiles", there is no concept of folders in Deta drives. There are just files with names. The Deta drive is the key-value storage, a bucket. The key is the file name, and the value is the file itself.

The file's name can have a prefix. For example, the name can look like "/abc/xyz/my_file.jpg". The "/abc/xyz" is the prefix of the name. It looks like a folder name, but it is not. 

You can list all the files in the drive via `list()` and then check the prefixes you are looking for.

To list the file with a given prefix, use the `prefix parameter of `list()`.

Finally, it isn't possible if you want to list all the drives programmatically in the project. You need to go to Deta UI and check there visually.

**Comment by esaid** [comment]
ok thank 's for your help.  i understand now the concept. 
cheers
Emmanuel
